### ISSUE TITLE
Error initializing Ray

### ISSUE THREAD

**Issue by devin-petersohn** [ray-project/ray#4193]
This issue was initially raised in Modin: modin-project/modin#468 by @carlosssanchez and @shaunstoltz

Logs from that issue pasted below for convenience:

> Thanks for quick reply. Have tried the code snippet above, and same error:

```
import ray

object_store_memory = int(0.6 * ray.utils.get_system_memory() // 10 ** 9 * 10 ** 9)
ray.init(
            include_webui=False,
            ignore_reinit_error=True,
            plasma_directory="/tmp",
            object_store_memory=object_store_memory,
        )
```

```
2019-02-27 23:08:59,437	INFO node.py:278 -- Process STDOUT and STDERR is being redirected to /tmp/ray/session_2019-02-27_23-08-59_73139/logs.
2019-02-27 23:08:59,545	INFO services.py:396 -- Waiting for redis server at 127.0.0.1:12520 to respond...
2019-02-27 23:08:59,659	INFO services.py:396 -- Waiting for redis server at 127.0.0.1:31524 to respond...
2019-02-27 23:08:59,662	INFO services.py:798 -- Starting Redis shard with 10.0 GB max memory.
2019-02-27 23:08:59,679	WARNING services.py:1242 -- WARNING: object_store_memory is not verified when plasma_directory is set.
2019-02-27 23:08:59,680	INFO services.py:1360 -- Starting the Plasma object store with 60.0 GB memory using /tmp.
---------------------------------------------------------------------------
ArrowIOError                              Traceback (most recent call last)
<ipython-input-1-4b0bccaf545a> in <module>()
      6             ignore_reinit_error=True,
      7             plasma_directory="/tmp",
----> 8             object_store_memory=object_store_memory,
      9         )

~/anaconda3/lib/python3.6/site-packages/ray/worker.py in init(redis_address, num_cpus, num_gpus, resources, object_store_memory, redis_max_memory, node_ip_address, object_id_seed, num_workers, local_mode, driver_mode, redirect_worker_output, redirect_output, ignore_reinit_error, num_redis_shards, redis_max_clients, redis_password, plasma_directory, huge_pages, include_webui, driver_id, configure_logging, logging_level, logging_format, plasma_store_socket_name, raylet_socket_name, temp_dir, _internal_config, use_raylet)
   1492         mode=driver_mode,
   1493         worker=global_worker,
-> 1494         driver_id=driver_id)
   1495 
   1496     for hook in _post_init_hooks:

~/anaconda3/lib/python3.6/site-packages/ray/worker.py in connect(info, redis_password, object_id_seed, mode, worker, driver_id)
   1800     # Create an object store client.
   1801     worker.plasma_client = thread_safe_client(
-> 1802         plasma.connect(info["store_socket_name"]))
   1803 
   1804     raylet_socket = info["raylet_socket_name"]

~/anaconda3/lib/python3.6/site-packages/ray/pyarrow_files/pyarrow/_plasma.pyx in pyarrow._plasma.connect()

~/anaconda3/lib/python3.6/site-packages/ray/pyarrow_files/pyarrow/error.pxi in pyarrow.lib.check_status()

ArrowIOError: Could not connect to socket /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store
```
$ python --version
Python 3.6.4 :: Anaconda custom (64-bit)
$ python -m pip list
alabaster (0.7.10)
anaconda-client (1.6.9)
anaconda-navigator (1.7.0)
anaconda-project (0.8.2)
asn1crypto (0.24.0)
astroid (1.6.1)
astropy (2.0.3)
attrs (17.4.0)
Babel (2.5.3)
backports.shutil-get-terminal-size (1.0.0)
beautifulsoup4 (4.6.0)
bitarray (0.8.1)
bkcharts (0.2)
blaze (0.11.3)
bleach (2.1.2)
blessings (1.7)
bokeh (0.12.13)
boto (2.48.0)
Bottleneck (1.2.1)
branca (0.3.0)
catamount (0.9)
certifi (2018.10.15)
cffi (1.11.4)
chardet (3.0.4)
click (6.7)
cloudpickle (0.5.2)
clyent (1.2.2)
colorama (0.3.9)
colorlover (0.2.1)
conda (4.5.11)
conda-build (3.4.1)
conda-verify (2.0.0)
contextlib2 (0.5.5)
cryptography (2.1.4)
cufflinks (0.14.6)
cycler (0.10.0)
Cython (0.27.3)
cytoolz (0.9.0)
dask (0.16.1)
datashape (0.5.4)
decorator (4.2.1)
distributed (1.20.2)
docutils (0.14)
entrypoints (0.2.3)
et-xmlfile (1.0.1)
evdev (1.1.2)
ez-setup (0.9)
fastcache (1.0.2)
filelock (2.0.13)
Flask (0.12.2)
Flask-Cors (3.0.3)
flatbuffers (1.10)
folium (0.5.0)
funcsigs (1.0.2)
gevent (1.2.2)
Glances (3.0.2)
glob2 (0.6)
gmpy2 (2.0.8)
gpustat (0.5.0)
greenlet (0.4.12)
h5py (2.7.1)
heapdict (1.0.0)
html5lib (1.0.1)
idna (2.6)
imageio (2.2.0)
imagesize (0.7.1)
ipykernel (4.8.0)
ipython (6.2.1)
ipython-genutils (0.2.0)
ipywidgets (7.1.1)
isort (4.2.15)
itsdangerous (0.24)
jdcal (1.3)
jedi (0.11.1)
Jinja2 (2.10)
jsonschema (2.6.0)
jupyter (1.0.0)
jupyter-client (5.2.2)
jupyter-console (5.2.0)
jupyter-core (4.4.0)
jupyterlab (0.31.5)
jupyterlab-launcher (0.10.2)
kiwisolver (1.0.1)
lazy-object-proxy (1.3.1)
llvmlite (0.21.0)
locket (0.2.0)
lxml (4.1.1)
MarkupSafe (1.0)
matplotlib (2.2.2)
mccabe (0.6.1)
mistune (0.8.3)
modin (0.3.1)
moviepy (0.2.3.5)
mpmath (1.0.0)
msgpack-python (0.5.1)
multipledispatch (0.4.9)
navigator-updater (0.1.0)
nbconvert (5.3.1)
nbformat (4.4.0)
networkx (2.1)
nltk (3.2.5)
nose (1.3.7)
notebook (5.4.0)
numba (0.36.2)
numexpr (2.6.4)
numpy (1.14.0)
numpydoc (0.7.0)
nvidia-ml-py3 (7.352.0)
odo (0.5.1)
olefile (0.45.1)
openpyxl (2.4.10)
packaging (16.8)
pandas (0.23.4)
pandas-datareader (0.6.0)
pandocfilters (1.4.2)
parso (0.1.1)
partd (0.3.8)
path.py (10.5)
pathlib2 (2.3.0)
patsy (0.5.0)
pep8 (1.7.1)
pexpect (4.3.1)
pickleshare (0.7.4)
Pillow (5.1.0)
pip (9.0.1)
pkginfo (1.4.1)
plotly (3.4.1)
pluggy (0.6.0)
ply (3.10)
prompt-toolkit (1.0.15)
psutil (5.4.7)
ptyprocess (0.5.2)
py (1.5.2)
pycodestyle (2.3.1)
pycosat (0.6.3)
pycparser (2.18)
pycrypto (2.6.1)
pycurl (7.43.0.1)
pyflakes (1.6.0)
Pygments (2.2.0)
pylint (1.8.2)
pyodbc (4.0.22)
pyOpenSSL (17.5.0)
pyparsing (2.2.0)
PySocks (1.6.7)
pytest (3.3.2)
python-dateutil (2.6.1)
pytz (2017.3)
pyusb (1.0.2)
PyWavelets (0.5.2)
PyYAML (3.12)
pyzmq (16.0.3)
QtAwesome (0.4.4)
qtconsole (4.3.1)
QtPy (1.3.1)
ray (0.6.3)
redis (3.2.0)
requests (2.18.4)
requests-file (1.4.3)
requests-ftp (0.3.1)
retrying (1.3.3)
rope (0.10.7)
ruamel-yaml (0.15.35)
scikit-image (0.13.1)
scikit-learn (0.19.1)
scipy (1.0.0)
seaborn (0.8.1)
Send2Trash (1.4.2)
setproctitle (1.1.10)
setuptools (38.4.0)
simplegeneric (0.8.1)
singledispatch (3.4.0.3)
six (1.11.0)
snowballstemmer (1.2.1)
sortedcollections (0.5.3)
sortedcontainers (1.5.9)
Sphinx (1.6.6)
sphinxcontrib-websupport (1.0.1)
spyder (3.2.6)
SQLAlchemy (1.2.1)
sqlalchemy-dremio (0.1.0.dev0)
statsmodels (0.8.0)
sympy (1.1.1)
tables (3.4.2)
tblib (1.3.2)
terminado (0.8.1)
testpath (0.3.1)
toolz (0.9.0)
tornado (4.5.3)
tqdm (4.23.4)
traitlets (4.3.2)
twitter (1.18.0)
typing (3.6.2)
unicodecsv (0.14.1)
urllib3 (1.22)
utils (0.9.0)
wcwidth (0.1.7)
webencodings (0.5.1)
Werkzeug (0.14.1)
wheel (0.30.0)
widgetsnbextension (3.1.0)
wrapt (1.10.11)
xlrd (1.1.0)
XlsxWriter (1.0.2)
xlwt (1.3.0)
zict (0.1.3)

```
[I 23:08:49.734 LabApp] Kernel restarted: 5e6a7278-bd80-4aec-9596-ed1d4eefae79
[I 23:08:50.335 LabApp] Adapting to protocol v5.1 for kernel 5e6a7278-bd80-4aec-9596-ed1d4eefae79
[I 23:08:50.336 LabApp] Restoring connection for 5e6a7278-bd80-4aec-9596-ed1d4eefae79:5a901346bdf87f0d0015cfd29c8056b2
[I 23:08:50.336 LabApp] Replaying 5 buffered messages
WARNING: Logging before InitGoogleLogging() is written to STDERR
E0227 23:08:59.697196 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 50 more times
WARNING: Logging before InitGoogleLogging() is written to STDERR
E0227 23:08:59.708341 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 50 more times
E0227 23:08:59.797468 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 49 more times
E0227 23:08:59.808454 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 49 more times
E0227 23:08:59.897589 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 48 more times
E0227 23:08:59.908550 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 48 more times
E0227 23:08:59.997697 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 47 more times
E0227 23:09:00.008649 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 47 more times
E0227 23:09:00.097797 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 46 more times
E0227 23:09:00.108742 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 46 more times
E0227 23:09:00.197921 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 45 more times
E0227 23:09:00.208851 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 45 more times
E0227 23:09:00.298036 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 44 more times
E0227 23:09:00.308956 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 44 more times
E0227 23:09:00.398152 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 43 more times
E0227 23:09:00.409060 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 43 more times
E0227 23:09:00.498260 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 42 more times
E0227 23:09:00.509160 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 42 more times
E0227 23:09:00.598383 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 41 more times
E0227 23:09:00.609275 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 41 more times
E0227 23:09:00.698513 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 40 more times
E0227 23:09:00.709403 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 40 more times
E0227 23:09:00.798641 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 39 more times
E0227 23:09:00.809523 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 39 more times
E0227 23:09:00.898766 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 38 more times
E0227 23:09:00.909641 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 38 more times
E0227 23:09:00.998893 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 37 more times
E0227 23:09:01.009758 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 37 more times
E0227 23:09:01.099022 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 36 more times
E0227 23:09:01.109884 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 36 more times
E0227 23:09:01.199153 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 35 more times
E0227 23:09:01.210002 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 35 more times
E0227 23:09:01.299286 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 34 more times
E0227 23:09:01.310122 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 34 more times
E0227 23:09:01.399394 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 33 more times
E0227 23:09:01.410223 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 33 more times
E0227 23:09:01.499480 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 32 more times
E0227 23:09:01.510315 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 32 more times
E0227 23:09:01.599562 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 31 more times
E0227 23:09:01.610406 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 31 more times
E0227 23:09:01.699648 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 30 more times
E0227 23:09:01.710501 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 30 more times
E0227 23:09:01.799743 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 29 more times
E0227 23:09:01.810600 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 29 more times
E0227 23:09:01.899832 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 28 more times
E0227 23:09:01.910693 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 28 more times
E0227 23:09:01.999919 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 27 more times
E0227 23:09:02.010782 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 27 more times
E0227 23:09:02.100001 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 26 more times
E0227 23:09:02.110865 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 26 more times
E0227 23:09:02.200103 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 25 more times
E0227 23:09:02.210973 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 25 more times
E0227 23:09:02.300210 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 24 more times
E0227 23:09:02.311079 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 24 more times
E0227 23:09:02.400315 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 23 more times
E0227 23:09:02.411183 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 23 more times
E0227 23:09:02.500421 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 22 more times
E0227 23:09:02.511282 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 22 more times
E0227 23:09:02.600519 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 21 more times
E0227 23:09:02.611387 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 21 more times
E0227 23:09:02.700619 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 20 more times
E0227 23:09:02.711488 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 20 more times
E0227 23:09:02.800721 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 19 more times
E0227 23:09:02.811589 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 19 more times
E0227 23:09:02.900823 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 18 more times
E0227 23:09:02.911689 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 18 more times
E0227 23:09:03.000928 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 17 more times
E0227 23:09:03.011793 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 17 more times
E0227 23:09:03.101037 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 16 more times
E0227 23:09:03.111896 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 16 more times
E0227 23:09:03.201154 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 15 more times
E0227 23:09:03.212009 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 15 more times
E0227 23:09:03.301265 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 14 more times
E0227 23:09:03.312115 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 14 more times
E0227 23:09:03.401366 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 13 more times
E0227 23:09:03.412215 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 13 more times
E0227 23:09:03.501466 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 12 more times
E0227 23:09:03.512312 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 12 more times
E0227 23:09:03.601572 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 11 more times
E0227 23:09:03.612416 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 11 more times
E0227 23:09:03.701681 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 10 more times
E0227 23:09:03.712527 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 10 more times
E0227 23:09:03.801787 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 9 more times
E0227 23:09:03.812635 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 9 more times
E0227 23:09:03.901895 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 8 more times
E0227 23:09:03.912739 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 8 more times
E0227 23:09:04.001994 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 7 more times
E0227 23:09:04.012836 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 7 more times
E0227 23:09:04.102088 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 6 more times
E0227 23:09:04.112932 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 6 more times
E0227 23:09:04.202205 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 5 more times
E0227 23:09:04.213044 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 5 more times
E0227 23:09:04.302312 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 4 more times
E0227 23:09:04.313140 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 4 more times
E0227 23:09:04.402408 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 3 more times
E0227 23:09:04.413235 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 3 more times
E0227 23:09:04.502508 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 2 more times
E0227 23:09:04.513332 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 2 more times
E0227 23:09:04.602599 73188 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 1 more times
E0227 23:09:04.613422 73139 io.cc:168] Connection to IPC socket failed for pathname /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store, retrying 1 more times
F0227 23:09:04.702714 73188 object_store_notification_manager.cc:23]  Check failed: _s.ok() Bad status: IOError: Could not connect to socket /tmp/ray/session_2019-02-27_23-08-59_73139/sockets/plasma_store
*** Check failure stack trace: ***
*** Aborted at 1551326944 (unix time) try "date -d @1551326944" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGABRT (@0x3e800011de4) received by PID 73188 (TID 0x7fd038fa0740) from PID 73188; stack trace: ***
    @     0x7fd038ba4390 (unknown)
    @     0x7fd037d55428 gsignal
    @     0x7fd037d5702a abort
    @           0x5b906d google::logging_fail()
    @           0x5b9096 google::LogMessage::Fail()
    @           0x5b8fe2 google::LogMessage::SendToLog()
    @           0x5b8966 google::LogMessage::Flush()
    @           0x5b8775 google::LogMessage::~LogMessage()
    @           0x503618 ray::RayLog::~RayLog()
    @           0x5633ed ray::ObjectStoreNotificationManager::ObjectStoreNotificationManager()
    @           0x51c5f3 ray::ObjectManager::ObjectManager()
    @           0x4c45d4 ray::raylet::Raylet::Raylet()
    @           0x4b3f97 main
    @     0x7fd037d40830 __libc_start_main
    @           0x4b9c91 (unknown)
```

**Comment by zbarry** [comment]
@devin-petersohn @carlosssanchez @shaunstoltz

No idea if it's related, but I encountered tmp directory issues in my case because I was operating on a cluster node with a buddy who was also using Ray (and had run Ray on the same physical node). Since Ray is coded to use /tmp/ray by default, if someone else has used Ray before you on your box, they're going to own that folder, and you/Ray are not going to be able to write to it. I just used `--temp-dir=blah` when launching my Ray workers / head node (and I think that's an option in `ray.init()` if you're just using the local parallelism). Never encountered the Arrow manifestation of this issue (if that's what it is; I haven't tried Modin, yet), but it seems plausible that this could be happening to you guys, too.
